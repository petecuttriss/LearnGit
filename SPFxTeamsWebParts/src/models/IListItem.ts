export interface ILinkItem {
    Description: string;
    Url: string;
}
export interface IListItem {
    Id: number;
    Title: string;
    SFDesc: string;
    SIAStatus: string;
    SFLink: ILinkItem;
    Function: string;
    Activity: string;
    Subactivity: string;
}
export interface IGroupedItems {
    Title: string;
    Items: IListItem[];
}