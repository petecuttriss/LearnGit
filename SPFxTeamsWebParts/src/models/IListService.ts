export interface IListService {
    getPropertyBagItem(item:string): Promise<string>;
    getItems(siteUrl: string, listId: string, fields: string, groupBy: string, filter?: string): Promise<any>;
    searchItems(query: string, siteUrl: string, listId: string, fields: string, groupBy: string, filter?: string): Promise<any>;
}