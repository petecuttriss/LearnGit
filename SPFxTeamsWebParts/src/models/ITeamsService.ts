export interface ITeamsService {
    getGroupId(): Promise<string>;
    getChannels(groupId: string): Promise<any[]>;
    getTabsFromChannel(groupId: string, channelId: string): Promise<any[]>;
    getMyTeams(): Promise<any[]>;
    getTeam(groupId: string): Promise<any>;
}