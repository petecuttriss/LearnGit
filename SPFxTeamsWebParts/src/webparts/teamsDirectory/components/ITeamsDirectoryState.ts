export interface ITeamsDirectoryState {
    loading: boolean;
    error: string;
    isSearch: boolean;
    items: any[];
}