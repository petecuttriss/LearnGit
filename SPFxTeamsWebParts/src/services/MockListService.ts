import { IListService } from "../models/IListService";
import { Utils } from "./Utils";

export class MockListService implements IListService {
    public async getPropertyBagItem(item: string): Promise<string> {
        var propertyBagItem: string = "Water Supply";

        return propertyBagItem;
    }

    public getItems(siteUrl: string, listId: string, fields: string, groupBy: string, filter?: string): Promise<any> {
        let mockItems: any = require("./items.json");
        let groupedItems = Utils.groupBy(mockItems, item => item[groupBy]);
        let sortedItems = Utils.sortByKey(groupedItems);

        return new Promise<any>((resolve) => {
            //setTimeout(() => {
            resolve(sortedItems);
            //}, 2000);
        });
    }

    public searchItems(query: string, siteUrl: string, listId: string, groupBy: string, filter?: string): Promise<any> {
        let mockItems: any = require("./items.json");
        let searchResults = mockItems.filter((item) => {
            return item.Title.toLowerCase().search(
                query.toLowerCase()) !== -1; //|| item.SFDesc.toLowerCase().search(query.toLowerCase()) !== -1;
        });

        let groupedItems = Utils.groupBy(searchResults, item => item[groupBy]);
        let sortedItems = Utils.sortByKey(groupedItems);
        return new Promise<any>((resolve) => {
            setTimeout(() => {
                resolve(sortedItems);
            }, 2000);
        });
    }
}