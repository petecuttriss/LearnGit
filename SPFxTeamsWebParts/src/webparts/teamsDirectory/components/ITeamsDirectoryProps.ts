import { DisplayMode } from "@microsoft/sp-core-library";
import { IListService } from "../../../models/IListService";

export interface ITeamsDirectoryProps {
  title: string;
  siteUrl: string;
  listId: string;
  hideTitle: boolean;
  hideSearchBox: boolean;
  openLinksInANewTab: boolean;
  displayMode: DisplayMode;
  listService: IListService;
  updateProperty: (value: string) => void;
  configureStartCallback: () => void;
}
