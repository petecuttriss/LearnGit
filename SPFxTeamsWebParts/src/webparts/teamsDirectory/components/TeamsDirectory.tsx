import * as React from 'react';
import styles from './TeamsDirectory.module.scss';
import { ITeamsDirectoryProps } from './ITeamsDirectoryProps';
import { escape, isEmpty } from '@microsoft/sp-lodash-subset';
import { ITeamsDirectoryState } from './ITeamsDirectoryState';
import { Nav, INavLinkGroup, INavLink } from 'office-ui-fabric-react';
import { Placeholder } from "@pnp/spfx-controls-react/lib/Placeholder";
import { IListItem } from '../../../models/IListItem';
import { DisplayMode } from '@microsoft/sp-core-library';
import { WebPartTitle } from '@pnp/spfx-controls-react/lib/WebPartTitle';
import { SearchBox } from 'office-ui-fabric-react';
import { MessageBar, MessageBarType } from 'office-ui-fabric-react';

const FIELDSTOSELECT: string = "Title,SFLink,ContentType,SIAStatus,SFDesc,Function,Activity,Subactivity";
const GROUPBY: string = "Activity";
const FILTER: string = "SIAStatus eq '03 Active'";

export default class TeamsDirectory extends React.Component<ITeamsDirectoryProps, ITeamsDirectoryState> {

  /**
   * Constructor
   * @param props 
   */
  constructor(props: ITeamsDirectoryProps) {

    super(props);

    this._onSearch = this._onSearch.bind(this);
    this._onSearchClear = this._onSearchClear.bind(this);
    this._configureWebPart = this._configureWebPart.bind(this);

    this.state = {
      error: null,
      loading: true,
      isSearch: false,
      items: []
    };
  }

  /**
   * componentDidMount
   * 
   * Get the list items from the sites register.
   */
  public componentDidMount() {
    this._getItems();
  }

  /**
   * Check the updated component props and if necessary get an updated result set.
   * @param prevProps 
   * @param prevState 
   */
  public componentDidUpdate(prevProps, prevState) {
    if (this.props !== prevProps) {
      if (this.props.siteUrl !== prevProps.siteUrl
        || this.props.listId !== prevProps.listId) {
        this.setState({
          loading: true,
          items: []
        });

        this._getItems();
      }
    }
  }

  /**
   * Render the component
   */
  public render(): React.ReactElement<ITeamsDirectoryProps> {
    const mandatoryFieldsConfigured = this._mandatoryFieldsConfigured();

    return (
      <div className={styles.teamsDirectory}>
        <div className={styles.container}>
          <div className={styles.row}>
            <div className={styles.column}>
              <div className={'ms-Grid'} dir="ltr">
                <div className={'ms-Grid-row'}>
                  {!mandatoryFieldsConfigured &&
                    <Placeholder iconName='Edit'
                      iconText='Configure your web part'
                      description='Please configure the web part.'
                      buttonLabel='Configure'
                      hideButton={this.props.displayMode === DisplayMode.Read}
                      onConfigure={this._configureWebPart} />
                  }
                  {mandatoryFieldsConfigured && !this.props.hideTitle &&
                    <div className='ms-Grid-col ms-sm12 ms-md7'>
                      <WebPartTitle className={styles.title} displayMode={this.props.displayMode}
                        title={this.props.title}
                        updateProperty={this.props.updateProperty} />
                    </div>
                  }
                  {mandatoryFieldsConfigured && !this.props.hideSearchBox &&
                    <div className='ms-Grid-col ms-sm12 ms-md5'>
                      <SearchBox onSearch={this._onSearch} onClear={this._onSearchClear} className={styles.searchbox} placeholder="Search by title" />
                    </div>
                  }
                </div>
                <div className={'ms-Grid-row'}>
                  <div className='ms-Grid-col ms-sm12'>
                    {mandatoryFieldsConfigured && this.state.error &&
                      <MessageBar messageBarType={MessageBarType.error}>{this.state.error}</MessageBar>
                    }
                    {mandatoryFieldsConfigured && Object.keys(this.state.items).length > 0 && !this.state.error &&
                      <Nav
                        groups={this.state.items}
                        collapsedStateText="Collapsed"
                        expandedStateText="Expand"
                        className={styles.nav}
                        key={Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15)}
                      />
                    }
                    {mandatoryFieldsConfigured && Object.keys(this.state.items).length == 0 && !this.state.error &&
                      <MessageBar messageBarType={MessageBarType.info}>No results found.</MessageBar>
                    }
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }

  /**
   * Return the list items from the configured Workplace Catalogue list
   */
  private _getItems() {
    if (this.props.siteUrl && this.props.listId) {
      // Use the filter values to query the sites registry for items matching this filter
      this.props.listService.getItems(this.props.siteUrl, this.props.listId, FIELDSTOSELECT, GROUPBY, FILTER).then((listItems) => {
        if (listItems.error) {
          this.setState({
            error: listItems.error,
            items: [],
            isSearch: false
          });
        } else {
          this._mapItems(listItems, true, false);
        }
      });
    }
  }

  /**
   * When a search string is entered query the list items by title and return results
   * @param event 
   */
  private _onSearch(event) {
    if (event === undefined ||
      event === null ||
      event === "") {
      this.props.listService.getItems(this.props.siteUrl, this.props.listId, FIELDSTOSELECT, GROUPBY, FILTER).then((listItems) => {
        this._mapItems(listItems, true, false);
      });
    }
    else {
      this.props.listService.searchItems(event, this.props.siteUrl, this.props.listId, FIELDSTOSELECT, GROUPBY, FILTER).then((listItems) => {
        this._mapItems(listItems, false, true);
      });
    }
  }

  /**
   * When the search is cleared, get the default configured list items
   * @param event 
   */
  private _onSearchClear(event) {
    this.props.listService.getItems(this.props.siteUrl, this.props.listId, FIELDSTOSELECT, GROUPBY, FILTER).then((listItems) => {
      this._mapItems(listItems, true, false);
    });
  }

  /**
   * Takes the returned list items and converts them to INavLinkGroup and INavLink items
   * @param groupItems 
   * @param collapse 
   * @param isSearch 
   */
  private _mapItems(groupItems: any[], collapse: boolean, isSearch: boolean): void {
    let tmpGroup: INavLinkGroup[] = [];
    Object.keys(groupItems).map((key, i) => {
      let tmpLinks: INavLink[] = [];
      groupItems[key].map((groupedItem: IListItem, index: number) => {
        tmpLinks.push({ key: groupedItem.Title, name: groupedItem.Title, url: groupedItem.SFLink === null ? "" : groupedItem.SFLink.Url, target: this.props.openLinksInANewTab ? "_blank" : "_self", disabled: groupedItem.SIAStatus !== "03 Active" });
      });
      tmpGroup.push({ name: key === "null" ? "Unknown" : key, links: tmpLinks, collapseByDefault: collapse });
    });

    this.setState({
      error: null,
      items: tmpGroup,
      isSearch: isSearch
    });
  }

  /**
   * Returns whether all mandatory fields are configured or not.
   * If the fields aren't configured we render the PlaceHolder component.
   */
  private _mandatoryFieldsConfigured(): boolean {
    return !isEmpty(this.props.siteUrl) &&
      !isEmpty(this.props.listId);
  }

  /**
   * Calls the parent web part to open the property pane
   */
  private _configureWebPart(): void {
    // Context of the web part
    this.props.configureStartCallback();
  }
}
