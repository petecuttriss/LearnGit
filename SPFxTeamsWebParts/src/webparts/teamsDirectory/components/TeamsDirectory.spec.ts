import * as React from 'react';
import 'jest';
import { shallow, mount, ReactWrapper, ShallowWrapper } from 'enzyme';
import TeamsDirectory from './TeamsDirectory';
import { MockListService } from '../../../services/MockListService';
import { IListService } from '../../../models/IListService';
import { DisplayMode } from "@microsoft/sp-core-library";
import { ITeamsDirectoryProps } from './ITeamsDirectoryProps';
import { ITeamsDirectoryState } from './ITeamsDirectoryState';

jest.mock('@microsoft/sp-core-library/', () => { return { 'BrowserDetection': {}, 'DisplayMode': {} }; });
jest.mock('@pnp/spfx-controls-react/lib/WebPartTitle', () => { return { 'WebPartTitle': 'WebPartTitle' }; });

describe('TeamsDirectory', () => {
  let wrapper: ShallowWrapper<ITeamsDirectoryProps, ITeamsDirectoryState>;
  let spy: jest.SpyInstance;
  
  beforeEach(() => {
    spy = jest.spyOn(TeamsDirectory.prototype, 'componentDidMount');
    wrapper = shallow(React.createElement(TeamsDirectory, {
      title: "Teams Directory",
      siteUrl: "/",
      listId: "1",
      hideTitle: false,
      hideSearchBox: false,
      openLinksInANewTab: true,
      displayMode: DisplayMode.Read,
      listService: new MockListService(),
      updateProperty: () => { },
      configureStartCallback: () => { }
    }));
  });

  test('should exist', () => {
    //const listServiceInstance: IListService = new MockListService();
    //const spy = jest.spyOn(TeamsDirectory.prototype, 'componentDidMount');
    /*const wrapper = shallow(React.createElement(TeamsDirectory, {
      title: "Teams Directory",
      siteUrl: "/",
      listId: "1",
      hideTitle: false,
      hideSearchBox: false,
      openLinksInANewTab: true,
      displayMode: DisplayMode.Read,
      listService: listServiceInstance,
      updateProperty: () => { },
      configureStartCallback: () => { }
    }));*/

    /*wrapper.setState({
      loading:false,
      error: null,
      isSearch: false,
      items: [
        {
          "odata.type": "SP.Data.SitesRegisterItem",
          "odata.id": "c805c25b-fef1-4273-97c8-11e71c7d86d8",
          "odata.etag": "\"4\"",
          "odata.editLink": "Web/Lists(guid'2b309a30-4a12-4167-a143-750601cffada')/Items(114)",
          "ContentType@odata.navigationLinkUrl": "Web/Lists(guid'2b309a30-4a12-4167-a143-750601cffada')/Items(114)/ContentType",
          "Title": "Auckland University Open Day",
          "SFDesc": null,
          "SIAStatus": "03 Active",
          "SFLink": {
            "Description": "Auckland University Open Day",
            "Url": "https://iworkplace.sharepoint.com/sites/AKLUNI"
          },
          "Function": "Corporate Management",
          "Activity": "Information Services",
          "Subactivity": null
        }]
    })*/
    expect(spy).toBeCalled();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('WebPartTitle').prop('title')).toBe("Teams Directory");
  });
});