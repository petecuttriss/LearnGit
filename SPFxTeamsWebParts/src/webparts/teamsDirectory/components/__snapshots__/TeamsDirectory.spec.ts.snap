// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamsDirectory should exist 1`] = `
<div
  className="teamsDirectory"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="ms-Grid"
          dir="ltr"
        >
          <div
            className="ms-Grid-row"
          >
            <div
              className="ms-Grid-col ms-sm12 ms-md7"
            >
              <WebPartTitle
                className="title"
                title="Teams Directory"
                updateProperty={[Function]}
              />
            </div>
            <div
              className="ms-Grid-col ms-sm12 ms-md5"
            >
              <StyledSearchBoxBase
                className="searchbox"
                onClear={[Function]}
                onSearch={[Function]}
                placeholder="Search by title"
              />
            </div>
          </div>
          <div
            className="ms-Grid-row"
          >
            <div
              className="ms-Grid-col ms-sm12"
            >
              <StyledNavBase
                className="nav"
                collapsedStateText="Collapsed"
                expandedStateText="Expand"
                groups={
                  Array [
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Cemeteries",
                          "name": "Cemeteries",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Cemeteries",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Communications",
                          "name": "Communications",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Communications",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Community Development",
                          "name": "Community Development",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Community Development",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Community Outcomes",
                          "name": "Community Outcomes",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Community Outcomes",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Recreation and Leisure",
                          "name": "Recreation and Leisure",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Community Services",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": false,
                          "key": "Contract Management",
                          "name": "Contract Management",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/contractmgmt",
                        },
                        Object {
                          "disabled": false,
                          "key": "Controlled Documents",
                          "name": "Controlled Documents",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/controlleddocs",
                        },
                      ],
                      "name": "Controlled Documents",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Customer Service",
                          "name": "Customer Service",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Customer Service",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Emergency Management",
                          "name": "Emergency Management",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Emergency Management",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Environmental Planning",
                          "name": "Environmental Planning",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Environmental Planning",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Executive Management",
                          "name": "Executive Management",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Executive Management",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "External Relationships",
                          "name": "External Relationships",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "External Relationships",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Governance",
                          "name": "Governance",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Governance",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Grants",
                          "name": "Grants",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Grants",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Health and Safety",
                          "name": "Health and Safety",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Health and Safety",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Housing",
                          "name": "Housing",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Housing",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": false,
                          "key": "Employee Files",
                          "name": "Employee Files",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/employeefiles",
                        },
                        Object {
                          "disabled": false,
                          "key": "HR Recruitment methods",
                          "name": "HR Recruitment methods",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/Reviewmarketingandpromotion",
                        },
                        Object {
                          "disabled": false,
                          "key": "Human Resources",
                          "name": "Human Resources",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/hr",
                        },
                        Object {
                          "disabled": false,
                          "key": "Recruitment",
                          "name": "Recruitment",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/recruitment",
                        },
                        Object {
                          "disabled": false,
                          "key": "Social Club Lunch",
                          "name": "Social Club Lunch",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/SocialClubLunch",
                        },
                        Object {
                          "disabled": false,
                          "key": "Staff remuneration benchmarking 2019-20",
                          "name": "Staff remuneration benchmarking 2019-20",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/Staffremunerationbenchmarking201920",
                        },
                        Object {
                          "disabled": false,
                          "key": "University of Auckland  Careers Day",
                          "name": "University of Auckland  Careers Day",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/UOACareers",
                        },
                      ],
                      "name": "Human Resources",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Information Management",
                          "name": "Information Management",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Information Management",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": false,
                          "key": "Auckland University Open Day",
                          "name": "Auckland University Open Day",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/AKLUNI",
                        },
                        Object {
                          "disabled": true,
                          "key": "Information Services",
                          "name": "Information Services",
                          "target": "_blank",
                          "url": "",
                        },
                        Object {
                          "disabled": false,
                          "key": "Massey University Open Day",
                          "name": "Massey University Open Day",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/MUniOPen",
                        },
                      ],
                      "name": "Information Services",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Libraries",
                          "name": "Libraries",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Libraries",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Management Accounting",
                          "name": "Management Accounting",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Management Accounting",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Noise Control",
                          "name": "Noise Control",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Noise Control",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Payments",
                          "name": "Payments",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Payments",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Rating",
                          "name": "Rating",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Rating",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Receivables",
                          "name": "Receivables",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Receivables",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Resource Consents",
                          "name": "Resource Consents",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Resource Consents",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Roading and Transport",
                          "name": "Roading and Transport",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Roading and Transport",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": false,
                          "key": "Team Admin",
                          "name": "Team Admin",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/TeamAdmin",
                        },
                        Object {
                          "disabled": false,
                          "key": "Team Management",
                          "name": "Team Management",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/TeamMgmt",
                        },
                      ],
                      "name": "Team Management",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Treasury",
                          "name": "Treasury",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Treasury",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": true,
                          "key": "Waste Management",
                          "name": "Waste Management",
                          "target": "_blank",
                          "url": "",
                        },
                      ],
                      "name": "Waste Management",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": false,
                          "key": "Bore head replacement",
                          "name": "Bore head replacement",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/BoreHeadReplacement",
                        },
                        Object {
                          "disabled": false,
                          "key": "Bromley Base Station Upgrade",
                          "name": "Bromley Base Station Upgrade",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/BromleyBaseStationUpgrade",
                        },
                        Object {
                          "disabled": false,
                          "key": "Bronson Bridge waterline repairs",
                          "name": "Bronson Bridge waterline repairs",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/BronsonBridge",
                        },
                        Object {
                          "disabled": false,
                          "key": "Burst Mains on High St - August 19",
                          "name": "Burst Mains on High St - August 19",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/BurstMainsonHighStAugust19",
                        },
                        Object {
                          "disabled": false,
                          "key": "Chlorination Comms Program",
                          "name": "Chlorination Comms Program",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/ChlorCommsProg",
                        },
                        Object {
                          "disabled": false,
                          "key": "Colombo street water mains earthquake repairs",
                          "name": "Colombo street water mains earthquake repairs",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/ColomboEQrepairs",
                        },
                        Object {
                          "disabled": false,
                          "key": "Eastern Suburb Flooding - June 19",
                          "name": "Eastern Suburb Flooding - June 19",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/EasternSuburbFloodingJune19",
                        },
                        Object {
                          "disabled": false,
                          "key": "Main Line Rebuild",
                          "name": "Main Line Rebuild",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/MainLineRebuild",
                        },
                        Object {
                          "disabled": false,
                          "key": "Merival Water Contamination Sep19",
                          "name": "Merival Water Contamination Sep19",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/MerivalWaterContaminationSep19",
                        },
                        Object {
                          "disabled": false,
                          "key": "Papanui Road Trunkline Comms",
                          "name": "Papanui Road Trunkline Comms",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/PapanuiRoadTrunklineComms",
                        },
                        Object {
                          "disabled": false,
                          "key": "SB165 Water Bore Replacement",
                          "name": "SB165 Water Bore Replacement",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/SB165",
                        },
                        Object {
                          "disabled": false,
                          "key": "Water Conservation Campaign",
                          "name": "Water Conservation Campaign",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/WaterConservation",
                        },
                        Object {
                          "disabled": false,
                          "key": "Water Supply",
                          "name": "Water Supply",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/watersupply",
                        },
                        Object {
                          "disabled": false,
                          "key": "Water Supply Communications",
                          "name": "Water Supply Communications",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/WaterSupplyCommunications",
                        },
                      ],
                      "name": "Water Supply",
                    },
                    Object {
                      "collapseByDefault": true,
                      "links": Array [
                        Object {
                          "disabled": false,
                          "key": "Canterbury University Open Day",
                          "name": "Canterbury University Open Day",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/CUNi",
                        },
                        Object {
                          "disabled": true,
                          "key": "Community Services",
                          "name": "Community Services",
                          "target": "_blank",
                          "url": "",
                        },
                        Object {
                          "disabled": false,
                          "key": "Corporate Management",
                          "name": "Corporate Management",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/corporatemgmt",
                        },
                        Object {
                          "disabled": false,
                          "key": "Corporate Netball Team",
                          "name": "Corporate Netball Team",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/Netball",
                        },
                        Object {
                          "disabled": false,
                          "key": "Example of a public site",
                          "name": "Example of a public site",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/PublicSiteExample",
                        },
                        Object {
                          "disabled": true,
                          "key": "Financial Management",
                          "name": "Financial Management",
                          "target": "_blank",
                          "url": "",
                        },
                        Object {
                          "disabled": false,
                          "key": "ILTestSite16883",
                          "name": "ILTestSite16883",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/ILTestSite16883",
                        },
                        Object {
                          "disabled": false,
                          "key": "Infrastructure Management",
                          "name": "Infrastructure Management",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/infrastructuremgmt",
                        },
                        Object {
                          "disabled": true,
                          "key": "Parks and Reserves",
                          "name": "Parks and Reserves",
                          "target": "_blank",
                          "url": "",
                        },
                        Object {
                          "disabled": false,
                          "key": "Projects",
                          "name": "Projects",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/projects",
                        },
                        Object {
                          "disabled": false,
                          "key": "Template - Activity Site",
                          "name": "Template - Activity Site",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/templateactivity",
                        },
                        Object {
                          "disabled": false,
                          "key": "Template - Project Site",
                          "name": "Template - Project Site",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/templateproject",
                        },
                        Object {
                          "disabled": false,
                          "key": "TestSIAEngine",
                          "name": "TestSIAEngine",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/TestSIAEngine",
                        },
                        Object {
                          "disabled": false,
                          "key": "Workspaces",
                          "name": "Workspaces",
                          "target": "_blank",
                          "url": "https://iworkplace.sharepoint.com/sites/Workspaces",
                        },
                      ],
                      "name": "Unknown",
                    },
                  ]
                }
                key="ezwsley6hh9gc4umv27ror"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
